uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/components/leave-type-wizard.component.js

```diff
@@ -84,6 +90,32 @@ define([
       return isHeaderOrElementInsideHeader;
     }
 
+    /**
+     * Checks if title provided is unique.
+     * If existing titles are not yet loaded, the validation will not be executed,
+     * instead, it will show that the field is "loading".
+     */
+    function checkIfTitleIsUnique () {
```

Comment:
Would not that execute all the accumulated `thens` once the promise is resolved considering checkIfTitleIsUnique should exec on each digest cycle?

Reply:
you are right so either

1) debounce it
2) 
```
if (!title.value || title.loading) {
  return true;
}
```

btw i would definitely change `title.loading` to `title.validating` given that it would be more descriptive of what it's actually going on when it it `true`

Reply:
I honestly think the current implementation is just fine and what is suggested may be a bit too much for this PR.

Reply:
I renamed the property.